# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Ravikiranbathe/flyrank-ai/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I chose the Refresh / Content Opportunity Scoring lane because I want to find which pages should be reviewed first for improvement. This lane looks useful because the starter dataset has different search and content signals that can help identify pages that may need attention.

In [13]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Question: Which pages should be reviewed first for possible improvement?

Decision: Decide which pages need attention first.

Action: A reviewer can check the suggested pages and decide whether to refresh, improve, or monitor them.

Cost of a wrong call: A wrong recommendation could waste time reviewing a page that does not need improvement, while a more important page may be missed.

In [14]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [15]:
import pandas as pd

url = "https://raw.githubusercontent.com/Ravikiranbathe/flyrank-ai/main/data/raw/content_refresh_anonymized.csv"
df = pd.read_csv(url)

print("Total rows:", len(df))
print("Declining pages:", (df["trend_direction"] == "down").sum())
print("Pages with 500+ impressions:", (df["impressions_90d"] >= 500).sum())

Total rows: 30000
Declining pages: 16262
Pages with 500+ impressions: 16726


The starter dataset has 30,000 pages. Out of these, 16,262 pages are declining and 16,726 pages have 500 or more impressions. This shows that there are many visible and declining pages, so it is useful to find which pages should be reviewed first.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This work can help identify pages that may need review based on the patterns observed in the data. It can support the decision of which pages to review first. However, it cannot prove that refreshing a page will improve its performance, and it cannot predict Google's algorithm.

In [16]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.